### PR TITLE
FEATURE: BB-343 Add totals to execution object

### PR DESCRIPTION
**Pull request by martinmosko**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/62/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/413/ : SUCCESS in 1m 11s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/59/ : SUCCESS in 1m 10s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/63/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/414/ : SUCCESS in 39s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/60/ : SUCCESS in 1m 00s (non-voting)

File: src/converters/tests/totals/totals.fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                ]
+            },
+            {
+                itemIdentifiers: ['measureGroup']
+            }
+        ]
+    }
+};
+
+export const withTotals: IVisualizationObjectContent = {
```
Comment (tomassatinsky) [inline src/converters/tests/totals/totals.fixtures.ts]: the name could be more descriptive

Comment (tomassatinsky) [reply src/converters/tests/totals/totals.fixtures.ts]: I see... it's part of another exported object in the end... then it doesn't need to be exported. But deserves better name anyway :)

File: src/converters/toAfmResultSpec.ts [hunk 2]
```diff
@@ -329,16 +353,48 @@ function generateDimensions(
         itemIdentifiersList[measureGroupIndex].push('measureGroup');
     }
 
-    return itemIdentifiersList.map(generateDimension);
+    return itemIdentifiersList.map(itemIdentifiers => generateDimension(itemIdentifiers, visObj, afm));
```
Comment (tomassatinsky) [inline src/converters/toAfmResultSpec.ts]: https://lodash.com/docs/4.17.4#partialRight can be used
`return itemIdentifiersList.map(partialRight(generateDimension, visObj, afm));`
Not sure if it's nicer :)

File: src/converters/tests/totals/totalsToAfmResultSpec.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    tables
+} from './totals.fixtures';
+
+import { TABLE } from '../../../constants/visualizationTypes';
+
+describe('totals toAfmResultSpec', () => {
+    const translatedPopSuffix = 'translated-pop-suffix';
+
+    it('should convert table grand totals', () => {
+        expect(toAfmResultSpec(tables.withTotals, translatedPopSuffix)).toEqual({
```
Comment (tomassatinsky) [inline src/converters/tests/totals/totalsToAfmResultSpec.spec.ts]: I'd prefer this to be split to two blocks - execution and assertion

File: src/converters/toAfmResultSpec.ts (hunk 2 above)
Comment (martinmosko) [reply src/converters/toAfmResultSpec.ts]: I would rather to keep it simple without using lodash :)

File: src/converters/tests/totals/totals.fixtures.ts (hunk 1 above)
Comment (martinmosko) [reply src/converters/tests/totals/totals.fixtures.ts]: changed

File: src/converters/tests/totals/totalsToAfmResultSpec.spec.ts (hunk 3 above)
Comment (martinmosko) [reply src/converters/tests/totals/totalsToAfmResultSpec.spec.ts]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/65/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/416/ : SUCCESS in 6m 15s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/62/ : SUCCESS in 1m 43s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-merge-compliance-check/25/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/417/ : SUCCESS in 52s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/63/ : SUCCESS in 2m 47s (non-voting)

**Pull request merged**